src/web/source.rs

```diff
@@ -198,7 +198,12 @@ pub fn source_browser_handler(req: &mut Request) -> IronResult<Response> {
 
     // get path (req_path) for FileList::from_path and actual path for super::file::File::from_path
     let (req_path, file_path) = {
-        let file_path = format!("sources/{}/{}/{}", crate_name, version, req_path.join("/"));
+        let mut req_path = req.url.path();
+        // remove first elements from path which is /crate/:name/:version/source
+        for _ in 0..4 {
+            req_path.remove(0);
+        }
+        let file_path = req_path.join("/");
```

Comment:
Hmm, are you sure this is correct? It will be missing `sources/`.

Reply:
`file_path` is the path in the source package, so having `sources/` here would be wrong.

Reply:
Oh I see, this changed because you changed `DbFile::from_path` to `fetch_source_file` below.